r1cs/gadgets/std/src/fields/mod.rs

```diff
@@ -239,6 +242,35 @@ pub trait FieldGadget<F: Field, ConstraintF: Field>:
         Ok(res)
     }
 
+    /// Computes `self^S`, where S is interpreted as an little-endian
+    /// u64-decomposition of an integer.
+    fn pow_by_constant<S: AsRef<[u64]>, CS: ConstraintSystem<ConstraintF>>(
+        &self,
+        mut cs: CS,
+        exp: S
+    ) -> Result<Self, SynthesisError> {
+        let mut res = Self::one(cs.ns(|| "alloc result"))?;
+        let mut found_one = false;
+
+        for i in BitIterator::new(exp) {
+
+            // Skip leading zeros
+            if !found_one {
+                if i {
+                    found_one = true;
+                } else {
+                    continue;
+                }
+            }
+            res.square_in_place(cs.ns(|| format!("square_{}", i)))?;
+
+            if i {
+                res.mul_in_place(cs.ns(|| format!("multiply_{}", i)), self)?;
+            }
+        }
+        Ok(res)
+    }
```

Comment:
To what extent we want to have this function to be secure in the context of incomplete arithmetics?
If so, I'd opt to implement the same strategy as for mul_bits, shifting the whole double and add by a random curve point.

Reply:
Why are you speaking about incomplete arithmetic ? This is exponentiation of a field element for a constant scalar